apps/hexagon_dma/process.cpp

```diff
@@ -46,23 +43,36 @@ int main(int argc, char **argv) {
 
     // We then need to prepare for copying to host. Attempting to copy
     // to host without doing this is an error.
-    halide_hexagon_dma_prepare_for_copy_to_host(nullptr, input, dma_engine);
+    // The Last parameter 0 indicate DMA Read
+    halide_hexagon_dma_prepare_for_copy_to_host(nullptr, input, dma_engine, false, eDmaFmt_NV12);
 
-    Halide::Runtime::Buffer<uint8_t> output(width, height);
+    Halide::Runtime::Buffer<uint8_t> output(width, height, 2);
+    Halide::Runtime::Buffer<uint8_t> output_y = output.cropped(2, 0, 1);    // Luma plane only
+    Halide::Runtime::Buffer<uint8_t> output_c = output.cropped(2, 1, 1).cropped(1, 0, (height/2));  // Chroma plane only, with reduced height
 
-    int result = pipeline(input, output);
+    int result = pipeline(input, output_y, output_c);
     if (result != 0) {
         printf("pipeline failed! %d\n", result);
     }
 
-    // Validate that the algorithm did what we expect.
-    output.for_each_element([&](int x, int y) {
-        uint8_t correct = input_validation(x, y) * 2;
-        if (correct != output(x, y)) {
-            printf("Mismatch at %d %d: %d != %d\n", x, y, correct, output(x, y));
-            abort();
+    output.copy_to_host();
+    int c = 2;
+    const int plane_start = 0;
+    const int plane_end = c;
```

Comment:
c -> 2, or better yet, just substitute in plane_start and plane_end, I don't see why these need to be variables.

Reply:
Agreed and corrected